File: tests/indirect-call.js
module.exports = [
  `i32 add(i32 a, i32 b) {
    return (a + b);
  };
  i32 sub(i32 a, i32 b) {
    return (a - b);
  };

  export i32 main() {
    i32 *func1 = add;
    if (1 == 1) { func1 = sub; }
    return (func1(10, 2));
  };
  `,
  `8`
];

File: test/indirect-call.c
int add(int a, int b) {
  return (a + b);
};

int sub(int a, int b) {
  return (a - b);
};

int main() {
  int (*func1)(int, int) = add;
  if (1 == 1) { func1 = sub; }
  assert(func1(10, 2), 8)
};
